FrEIA/modules/gaussian_mixture.py

```diff
@@ -146,35 +146,19 @@ def forward(self, x, c, rev=False):
         if not rev:
             if fixed_components:
                 # Return latent codes of x according to chosen component distributions only
-                return [torch.stack([torch.matmul(U[b,i[b],:,:], x[0][b,:] - mu[b,i[b],:]) for b in range(batch_size)])]
+                return [torch.stack([torch.matmul(U[b,i[b],:,:], x[0][b,:] - mu[b,i[b],:]) for b in range(batch_size)])], self.jac
             else:
                 # Return latent codes of x according to all component distributions simultaneously
                 if len(x[0].shape) < 3:
                     x[0] = x[0][:,None,:]
-                return [torch.matmul(U, (x[0] - mu)[...,None])[...,0]]
+                return [torch.matmul(U, (x[0] - mu)[...,None])[...,0]], self.jac
```

Comment:
Why self.jac? Jacobians should never be stored

Reply:
in the cases when it's constant, it makes sense to store it!

Reply:
Agreed.
Then we should have some code that says `if self.jac is not None: jac = ...`, but I only see `self.jac = ...`